FreeRTOS/FreeRTOS-Plus-TCP
# Check minimum size of ICMPv6 packets

Description
-----------
This PR will correct the minimum length check for ICMP/ping packets.

The current code assumes that an ICMPv6 packet always has a length of 24 or more: [`ipSIZE_OF_ICMPv6_HEADER`](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/source/include/FreeRTOS_IPv6.h#L43). In fact, echo packets have an ICMP header of 8 bytes only. The header may be followed by zero or more user data.

Test Steps
-----------
I sent pings of increasing size from 0 up to to 1452 bytes, where my MTU was 1500.
~~~
    ping -c 1 -s 0 fe80::8008
    ping -c 1 -s 1 fe80::8008
    ...
    ping -c 1 -s 1452 fe80::8008  // Results in a packet of 1500 + 14 bytes.
~~~

Checklist:
----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
